### DISCUSSION TITLE
Module to add items to the headtag of Chyrp Lite

### DISCUSSION THREAD

**Discussion by eawmaas** [xenocrat/chyrp-lite#199] [Show and tell]
I have been working all day yesterday and I was só close: no "file not found" errors or other errors anymore (after some help from Daniel) and it looked like it was going to work, but all it did was render a blank page. I wanted to make a mod where people can add scripts in the admin-section and I díd read the route-info several times but did not get the result that I wanted. I also tried adding settings to the config.json-file automatically, which worked, but still: a blank page. So I went to bed.

This morning I decided to use a different approach, realising that I also needed a way to store the scripts that users add and this different approach opened several other possibilities too, because the head-tag is not only useful for scripts but for stylesheets and SEO as well. So I let myself go a bit and created a 3 in 1 module where you can add scripts, styles(heets) and SEO-optimalisation. It is a bit traditional PHP but I just wanted to have a start with something and apply these items to my own Blog without changing the source-code of Chyrp Lite, which was my intention in the first place. IMHO the style-functionality is awesome, it brings a huge amount of possibilities with it.

So: this module is designed for highly- and not-so-highly- skilled developers, basically everybody can work with it and adjust it to his´ / hers´ own needs. Later-on I will dive into making it more sophisticated / elegant using the Chyrp-way of doing things (I am just not that far in my knowledge of Chyrp at the moment and I want to close this project for now and produce something that works). But I think this is a nice start and I think that there might be people who can be happy with this mod, well: I hope so.

I also added some niceties (as you call them) to it.

The Mod is in the ZIP, I hope you like it. I tested it on PHP 8.2
[headadditions.zip](https://github.com/xenocrat/chyrp-lite/files/12516124/headadditions.zip)

**Comment by xenocrat** [comment]
Thank you for sharing! I will be sure to download it and take a look.

**Reply by eawmaas** [reply]
Have fun, I know I do, it's a toy :-)
